src/Lorentz/Contracts/BaseDAO/Types.hs

```diff
@@ -211,7 +212,7 @@ data Storage (contractExtra :: Kind.Type) (proposalMetadata :: Kind.Type) = Stor
 
   , sExtra                     :: contractExtra
   , sProposals                 :: BigMap (ProposalKey proposalMetadata) (Proposal proposalMetadata)
-  , sProposalKeyListSortByDate :: [ProposalKey proposalMetadata] -- Newest first
+  , sProposalKeyListSortByDate :: Map (Timestamp, ProposalKey proposalMetadata) () -- Oldest first
```

Comment:
It looks isomorphic to a set, why are we using a map here?

Reply:
Yes, it is isomorphic to a set. This is due to, initially the map was `Map Timestamp ( ProposalKey proposalMetadata)`, but due to key collision problem, it becomes `Map (Timestamp, ProposalKey proposalMetadata) ()`.

Do you think we should change to `Set` or not.

Reply:
If we can change it to `Set`, I think we should do it. It should be more efficient I guess (and we are already close to hitting the operation size limit). Also it adds clarity.

Reply:
I've updated this now.